internal/pkg/agent/cmd/enroll_cmd.go

```diff
@@ -545,6 +550,9 @@ func (c *enrollCmd) enroll(ctx context.Context, persistentConfig map[string]inte
 		// use internal URL for future requests
 		if c.options.InternalURL != "" {
 			fleetConfig.Client.Host = c.options.InternalURL
+			// fleet-server will bind the internal listenter to localhost:8221
+			// InternalURL is localhost:8221, however cert uses $HOSTNAME, so we need to disable verification.
+			fleetConfig.Client.Transport.TLS.VerificationMode = tlscommon.VerifyNone
```

Comment:
Should Fleet Server use a different cert for it's internal port? Ideally, we'd make it impossible for a malicious actor to spin up a separate Fleet Server on the same host/port to control this agent.

Reply:
@joshdover I've changed it to verifying the certificate instead